fe1-web/model/network/method/message/data/MessageData.ts

```diff
@@ -35,3 +35,18 @@ export interface MessageData {
   readonly object: ObjectType;
   readonly action: ActionType;
 }
+
+/**
+ * Returns if the message data is signed with token or not. If not, it has to be signed with
+ * a public key.
+ *
+ * @remarks
+ * This function has to be updated for each new kind of message that is going to be signed using
+ * the pop token.
+ *
+ * @param data - The message data we want to know how to sign
+ */
+export function isSignedWithToken(data: MessageData): Boolean {
+  return (data.object === ObjectType.ELECTION && data.action === ActionType.CAST_VOTE)
+    || (data.object === ObjectType.CHIRP && data.action === ActionType.ADD);
+}
```

Comment:
to be discussed: would it be simpler to maintain if this were a property within the `data` itself?

Reply:
I think it would, because having to update this function every time we add a new type of message that needs to be signed with a token doesn't seem really practical. But to be honest, I don't know how to do it.